crates/sui-types/src/error.rs

```diff
@@ -59,42 +59,142 @@ macro_rules! exit_main {
+    MoveObjectAsPackage { object_id: ObjectID },
+    #[error("Transaction was not signed by the correct sender: {}", error)]
+    IncorrectUserSignature { error: String },
+
+    #[error("Object used as shared is not shared.")]
+    NotSharedObjectError,
+    #[error("The transaction inputs contain duplicated ObjectRef's")]
+    DuplicateObjectRefInput,
+
+    // Gas related errors
+    #[error("Gas object is not an owned object with owner: {:?}.", owner)]
+    GasObjectNotOwnedObject { owner: Owner },
+    #[error("Gas budget: {:?} is higher than max: {:?}.", gas_budget, max_budget)]
+    GasBudgetTooHigh { gas_budget: u64, max_budget: u64 },
+    #[error("Gas budget: {:?} is lower than min: {:?}.", gas_budget, min_budget)]
+    GasBudgetTooLow { gas_budget: u64, min_budget: u64 },
+    #[error(
+        "Balance of gas object {:?} is lower than gas budget: {:?}, with gas price: {:?}.",
+        gas_balance,
+        gas_budget,
+        gas_price
+    )]
+    GasBalanceTooLowToCoverGasBudget {
+        gas_balance: u128,
+        gas_budget: u128,
+        gas_price: u64,
+    },
+    #[error("Transaction kind does not support Sponsored Transaction")]
+    UnsupportedSponsoredTransactionKind,
+    #[error(
+        "Gas price {:?} under reference gas price (RGP) {:?}",
+        gas_price,
+        reference_gas_price
+    )]
+    GasPriceUnderRGP {
+        gas_price: u64,
+        reference_gas_price: u64,
+    },
+    #[error("Object {object_id} is not a gas object")]
+    InvalidGasObject { object_id: ObjectID },
+    #[error("Gas object does not have enough balance to cover minimal gas spend")]
+    InsufficientBalanceToCoverMinimalGas,
+
+    #[error("Could not find the referenced object {:?} as the asked version {:?} is higher than the latest {:?}", object_id, asked_version, latest_version)]
+    ObjectSequenceNumberTooHigh {
+        object_id: ObjectID,
+        asked_version: SequenceNumber,
+        latest_version: SequenceNumber,
+    },
+    #[error("Object deleted at reference {:?}.", object_ref)]
+    ObjectDeleted { object_ref: ObjectRef },
+    #[error("Invalid Batch Transaction: {}", error)]
+    InvalidBatchTransaction { error: String },
+    #[error("This Move function is currently disabled and not available for call")]
+    BlockedMoveFunction,
+    #[error("Empty input coins for Pay related transaction")]
+    EmptyInputCoins,
+
+    #[error("SUI payment transactions use first input coin for gas payment, but found a different gas object.")]
+    UnexpectedGasPaymentObject,
+
+    #[error("Wrong initial version given for shared object")]
+    SharedObjectStartingVersionMismatch,
+
+    #[error("Attempt to transfer object {object_id} that does not have public transfer. Object transfer must be done instead using a distinct Move function call.")]
+    TransferObjectWithoutPublicTransferError { object_id: ObjectID },
+
+    #[error("Feature is not yet supported: {0}")]
+    Unsupported(String),
+}
```

Comment:
There's probably a few more that need to be added here, e.g. the new TransactionExpired error case (although idk if this is on too old of a base)

Reply:
yeah we should migrate that over too